x/feemarket/post/fee_test.go

```diff
@@ -53,7 +54,52 @@ func TestDeductCoins(t *testing.T) {
 				s.MockBankKeeper.On("SendCoinsFromAccountToModule", s.Ctx, acc.Account.GetAddress(), types.FeeCollectorName, tc.coins).Return(nil).Once()
 			}
 
-			if err := post.DeductCoins(s.MockBankKeeper, s.Ctx, acc.Account, tc.coins); (err != nil) != tc.wantErr {
+			if err := post.DeductCoins(s.MockBankKeeper, s.Ctx, acc.Account, tc.coins, false); (err != nil) != tc.wantErr {
+				s.Errorf(err, "DeductCoins() error = %v, wantErr %v", err, tc.wantErr)
+			}
+		})
+	}
+}
+
+func TestDeductCoinsAndDistribute(t *testing.T) {
```

Comment:
There should also be an end to end test in place where we are checking the coins are distributed properly without using mocks